packages/react-components/react-platform-adapter/src/styling/convertGriffelToStyleX.native.ts

```diff
@@ -1,85 +1,131 @@
+): string | number | ValueWithSelectors {
+  if (selectors.length === 0) {
+    return value;
+  }
+
+  const selector = selectors[0];
+  const nestedSelectors = selectors.slice(1);
+
+  if (isValueWithSelectors(existingValue)) {
+    existingValue[selector] = makeValueWithSelectors(nestedSelectors, value, existingValue[selector]);
+    return existingValue;
+  }
+
+  return {
+    default: existingValue ?? null,
+    [selector]: makeValueWithSelectors(nestedSelectors, value),
+  } as ValueWithSelectors;
 }
 
-function mergeSelectorValues(
-  selector: keyof ComplexValueType,
-  style: GriffelStyle | GriffelResetStyle,
-  result: StyleXStyleBuilder,
-) {
-  for (const key in style) {
-    if (!Object.prototype.hasOwnProperty.call(style, key)) {
+/**
+ * Converts from Griffel's selector format to StyleX's format.
+ *
+ * Griffel nests the values inside of the selectors:
+ * {
+ *   color: 'red',
+ *  ':hover': {
+ *     color: 'green',
+ *     '@media (forced-colors: active)': {
+ *        color: 'highlighttext',
+ *      },
+ *   },
+ * }
+ *
+ * StyleX nests the selectors inside of the values:
+ * {
+ *   color: {
+ *     default: 'red',
+ *     ':hover': {
+ *       default: 'green',
+ *       '@media (forced-colors: active)': 'highlighttext',
+ *     },
+ *   },
+ * }
+ *
+ * @param selector The selector, such as ':hover' or '@media ...'
+ * @param griffelStyle The griffel styles within this selector.
+ * @param strictDomStyle The react-strict-dom style to add the merged values to.
+ */
+function mergeSelectorValues(selectors: Selector[], griffelStyle: GriffelStyle, strictDomStyle: StyleXStyleBuilder) {
+  for (const key in griffelStyle) {
+    if (!Object.prototype.hasOwnProperty.call(griffelStyle, key)) {
       continue;
     }
 
-    const value = style[key as keyof GriffelStyle];
-    if (typeof value === 'string' || typeof value === 'number') {
-      const property = key as keyof GriffelStyle & keyof StyleXStyle;
+    const property = key as keyof GriffelStyle & keyof StyleXStyle;
+    const value = griffelStyle[property];
 
-      // If there's already a value with another selector, add this to it
-      const existingValue = result[property];
-      if (isComplexValue(existingValue)) {
-        existingValue[selector] = value;
-      } else {
-        result[property] = { default: existingValue || null, [selector]: value };
-      }
+    if (typeof value === 'string' || typeof value === 'number') {
+      strictDomStyle[property] = makeValueWithSelectors(selectors, value, strictDomStyle[property]);
+    } else if (Array.isArray(value)) {
+      console.warn('Unsupported array value for ', key);
+    } else if (isSelector(key) && isValueWithSelectors(value)) {
+      mergeSelectorValues([...selectors, key], value, strictDomStyle);
     } else {
-      throw new Error(`Unsupported nested selector: '${selector}' '${key}'`);
+      console.warn(`Unsupported selector `, [...selectors, key]);
```

Comment:
If I follow the logic now, we will emit warnings with unsupported selectors but not end up passing them to stylex. I think that is better than having the hard errors.

Reply:
Yeah, we'll need a longer-term solution to catch and fix these cases at development time, like a lint rule or something. But for now, this at least makes it so it doesn't crash with an invalid selector.